src/libraries/System.Net.Http/src/System/Net/Http/WasiHttpHandler/WasiHttpHandler.cs

```diff
@@ -54,15 +54,14 @@ public async Task<HttpResponseMessage> SendRequestAsync(HttpRequestMessage reque
                 cancellationToken.ThrowIfCancellationRequested();
 
                 var response = new HttpResponseMessage((HttpStatusCode)incomingResponse.Status());
```

Comment:
Please also make sure to set the `RequestMessage` property of the response to the request. Like done in https://github.com/dotnet/runtime/pull/42822 for browser and to avoid issues like https://github.com/dotnet/runtime/issues/42691.

```suggestion
                var response = new HttpResponseMessage((HttpStatusCode)incomingResponse.Status());
                response.RequestMessage = request;
```